deep_agent/src/infrastructure/backend.py

```
@@ -285,7 +392,7 @@ def _build_state_backend() -> Any:
         from deepagents.backends.state import StateBackend
 
         logger.info("Using StateBackend (thread-scoped scratch)")
-        return StateBackend()
+        return lambda runtime: StateBackend(runtime)
```

Ruff (0.16.2)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)